src/main/scala/BattleShips/MainGame.scala

```diff
@@ -0,0 +1,36 @@
+package BattleShips
+
+object MainGame{
+
+  def takeUserInput(playerName: String , xOrYCords: String) : Int = {
+    println(s"Player $playerName Its Your Turn To Choose Your $xOrYCords Coordinates")
+    val playerInput = scala.io.StdIn.readInt()
+
+    val playerInputChecked = validateUserInputAgainstGridSize(playerInput)
```

Comment:
val playerInputChecked = validateUserInputAgainstGridSize(playerInput)
saving player input seems pointless without reusing it
you could
val playerInputChecked = validateUserInputAgainstGridSize(scala.io.StdIn.readInt())